doc/development/style_guide.md

```diff
@@ -0,0 +1,211 @@
+# Flame Style Guide
+
+This is a general style guide for writing code within Flame and adjacent projects. We strive to
+maintain the code clean and readable -- both for the benefit of the users who will need to study
+this code in order to understand how a particular feature works or debug something that is breaking,
+and for the benefit of the current and future maintainers.
+
+This guide extends upon the official style guides of Dart and Flutter. Those guides should be
+considered an essential part of this style guide. However, whenever some rules contradict others,
+the following order of preferences apply:
+- **Flame Style Guide** (this document)
+- **[Flutter Style Guide]**
```

Comment:
This one is the internal style guide for the Flutter project as far as I'm aware and shouldn't be included here.

Reply:
Alright, I'll remove the Flutter Style Guide then.